### ISSUE TITLE
Glastonbury is now detecting this script as a BOT and tabs crashing

### ISSUE THREAD

**Issue by jltwheeler** [JackOHara/glasto-helper#4]
Only a few of my tabs are successful in making a request to https://glastonbury.seetickets.com/content/extras/ with the others showing the response below:

![Screenshot 2024-10-26 at 16 51 07](https://github.com/user-attachments/assets/1e734b16-2a2f-4dbd-af66-e0f74eed97ba)

I'm also noticing a lot of tabs crashing constantly, which I find odd on an Macbook Pro (Apple M3 Pro chip with 18gb RAM).

**Comment by jltwheeler** [comment]
![image](https://github.com/user-attachments/assets/00cfa07e-d10c-4cb0-9381-ac7aa4b91c52)

And also this as soon as tabs increase over 10 in parallel

**Comment by andythornley** [comment]
I think the days of refresh bots are behind us unfortunately. We now have a queuing system...

https://glastonburyfestivals.co.uk/news/2025-ticket-sale-faq/

**Comment by jltwheeler** [comment]
lol just saw this too... RIP 

@andythornley is there anyway around this using proxy servers? there must be a way around this... plenty of people still create BOTs for other queuing websites e.g. ticketmaster, asx which all have queuing systems in place

**Comment by andythornley** [comment]
i would imagine some residential proxies would still work, but this news has thrown all my plans into disarray

**Comment by andythornley** [comment]
> lol just saw this too... RIP
> 
> @andythornley is there anyway around this using proxy servers? there must be a way around this... plenty of people still create BOTs for other queuing websites e.g. ticketmaster, asx which all have queuing systems in place

so i was thinking about using a similar solution to create new tabs but for each tab using a residential proxy and randomising the user agent

**Comment by bizarrist** [comment]
Stumbled on these comments as a GitHub tourist looking for Glasto ticket tips, given the new circumstances... Is any of what you are thinking above likely to be scriptable for public consumption, or are you just going to experiment by yourselves for now? Thx

**Comment by jltwheeler** [comment]
I was going to take a look this weekend actually. was thinking of doing something similar with each new tab having a different IP via a proxy. Were you going to build this yourself or use an off the shelf product? Looks like there are also some nice free ones to use online

**Comment by andythornley** [comment]
Yeah I was going to spend the weekend working on it as well, we could share our findings with each other?

**Comment by bizarrist** [comment]
I will be very little use other than being a test pilot if you want me to...?!

**Comment by joshuagornall** [comment]
you will have to pay for residential proxies, I looked into this and its really not to hard to solve with what you've mentioned above using residential proxies and a distributed system with puppeteer but its gonna cost about 6 euro per proxy.

**Comment by rorymullan** [comment]
You really think they'll be tracking proxies through VPN services like Nord? I'd be surprised. I'm going to give this a go from around the globe.

**Comment by keat-s** [comment]
I was trying something I put together using claude based on this: https://github.com/ahmedmujtaba1/Queue-Checker-Python

My version used playwright and thee zenrows (Free trial) API which is apparently good enough to beatk akamai's servers (where the new queue is being hosted) https://www.zenrows.com/blog/bypass-akamai#behavior-analysis 

I don't have the first clue about webdev, or scraping (usually powershell) but the browser was opening, and some even returned with a 200 response from the log. After several iterations trying in today's queue I wasn't able to get more than one tab open. I can share the project if you like to see where I might be going wrong? 

Did hefty research on how the new queue flow works, so if your interested in that doc let me know to. In short its about having the right cookie in your browser session after you've waited in the queue. apparently there's no way to spoof that, so many tabs, all with different IPs through a proxy seems like the way.

**Comment by keat-s** [comment]
> you will have to pay for residential proxies, I looked into this and its really not to hard to solve with what you've mentioned above using residential proxies and a distributed system with puppeteer but its gonna cost about 6 euro per proxy.

Did you try this? Any difference between using puppeteer/ selenium/ playwright. Apparently there's people on twitter saying the refresh method worked as well

**Comment by andythornley** [comment]
Yes I tried with puppeteer, the problem was that they use proof of work which seems to hammer data and cpu making residential proxies expensive.

Did anyone get the url for the ticket page,?

**Comment by keat-s** [comment]
I got the URL for page when countdown timer starts, and when you're in the queue. Got through on one of my normal tabs and was to stressed to get the URL for that

**Comment by keat-s** [comment]
> Yes I tried with puppeteer, the problem was that they use proof of work which seems to hammer data and cpu making residential proxies expensive.
> 
> Did anyone get the url for the ticket page,?

did you use glasto-helper with proxies then?

Just seen someone post on twitter sending a Get request to the API, and it shows you expected wait time in the queue. Maybe could incorporate that into the logic, and then open tab with lowest wait time? https://x.com/lawrjones/status/1857123886523163086?s=46

**Comment by andythornley** [comment]
> I got the URL for page when countdown timer starts, and when you're in the queue. Got through on one of my normal tabs and was to stressed to get the URL for that

Check your history should be able to get the url from there

**Comment by keat-s** [comment]
Url in waiting room: 
https://glastonbury.seetickets.com/queue/view?c=seeticketsuk&e=20241114glastocoach&ver=javascript-4.3.0&cver=68&man=Glastonbury%20Festival&t=https%3A%2F%2Fglastonbury.seetickets.com%2Fcontent%2Fextras&kupver=akamai-4.3.0


Some sort of tokenised url - see akamai edgeconnector docs for what this means
https://glastonbury.seetickets.com/queue/view?c=seeticketsuk&e=20241114glastocoach&ver=javascript-4.3.0&cver=68&man=Glastonbury%20Festival&enqueuetoken=eyJ0eXAiOiJRVDEiLCJlbmMiOiJBRVMyNTYiLCJpc3MiOjE3MzE2MDQ0NDA4MDQsInRpIjoiYjc5ODZkMzktZDg2My00ZThhLTg5YjYtNjkwZTMyZDI3OGY4IiwiYyI6InNlZXRpY2tldHN1ayIsImUiOiIyMDI0MTExNGdsYXN0b2NvYWNoIiwiaXAiOiI4Mi4xMzMuMi4yMiJ9.p2yC8dMyDVyA25iOLkLDMw.EU-SA5j2l9nciC_6rBNcPxSeS2zfBhjFT3Kx9TPitmE&t=https%3A%2F%2Fglastonbury.seetickets.com%2Fcontent%2Fextras&kupver=akamai-4.3.0

After getting through queue: - didnt sucessfully get to payment:
https://glastonbury.seetickets.com/event/glastonbury-2025-ticket-coach-travel/worthy-farm/3186906

**Comment by andythornley** [comment]
You absolute legend.  We should be able to work out the url for Sunday based on previous years and work with that.

**Comment by keat-s** [comment]
> You absolute legend. We should be able to work out the url for Sunday based on previous years and work with that.

no worries, let me know if you get anywhere, if you got any tips on where to go w my script that would be much appreciated, or even what kind of script to write. Our main group got in, but i bottled it for the other group with the bus page (which is horrendous)

**Comment by andythornley** [comment]
Post a link to it mate, we can work on things together

**Comment by keat-s** [comment]
sent invite mate, going to write a new one using glasto-helper.  See On original repo of glasto-helper > the guys blog, right at the bottom guy said that he had improved it using a proxy system, but file is private

**Comment by rorymullan** [comment]
deleting

**Comment by keat-s** [comment]
sorry rory, been at work - inviting you to private repo now; post your updates in there?

**Comment by linusjody** [comment]
@keat-s do you mind if I get involved with this?

**Comment by orsxn** [comment]
hi, new here. Was speaking to a guy last night who posted this on reddit at about 9am on Sunday.
https://docs.google.com/spreadsheets/d/1QyetvkeMuS4PlmjJJSC3lFBwB0ye_8j1jPDMj624Wrk/edit?gid=0#gid=0


Basically he said he was using requests to get the url's from the server that passes the next in line to the registration page. 

Anyone have an idea how this is done as I didn't even wake up for the sale.

original reddit thread - https://www.reddit.com/user/Medium_Willow_3727/submitted/

**Comment by keat-s** [comment]
Yea saw this too Orson, tried messaging him but he’s not very helpful/ clear on what his method or willing to share his process - this would be the golden automation script imo but he’s Gate keeping - 

UPDATE - I’m also not clear on whether what he’s posted are links that have gone through the queue - I.e is what he’s posted only a few  successful links from a massive number of requests to the site, or he’s used automation to somehow get the cookie that signed your queue id as having been through the queue.

**Comment by orsxn** [comment]
from what i read there were being pubished in real time, one person commented and said there were only 10 links in the spreadsheet when he had success.

**Comment by orsxn** [comment]
I think this method is being used alot by shoe resellers etc. Don't think many will want to release how this is being done.

**Comment by orsxn** [comment]
https://github.com/ganhj99/queueit-bot

Could be similar to what this guy is offering?

**Comment by andythornley** [comment]
> https://github.com/ganhj99/queueit-bot
> 
> Could be similar to what this guy is offering?

I think that's exactly what they are using

**Comment by orsxn** [comment]
i've been looking at shoe bots, alot have a queue it module. Look at stellaraio!

**Comment by orsxn** [comment]
keats could i join your repo? I'm trying with playwright

**Comment by keat-s** [comment]
> i've been looking at shoe bots, alot have a queue it module. Look at stellaraio!

Lmao might get into the shoe flip market if these bots are any good, recession hitting hard 😬

**Comment by orsxn** [comment]
you can rent them from whop, i was maybe thinking renting one for resale as it a queue it module. only like £6 a day.

**Comment by keat-s** [comment]
Added you to repo,

> you can rent them from whop, i was maybe thinking renting one for resale as it a queue it module. only like £6 a day.

Nice - Added you to playwright repo -  Send us a link to that in private repo?

**Comment by TheMole442** [comment]
> you can rent them from whop, i was maybe thinking renting one for resale as it a queue it module. only like £6 a day.


which one is it that you can rent ? I messaged one of the guys you linked earlier in the thread about pricing

**Comment by orsxn** [comment]
i was looking a stellar aio, i put the wrong page. That one is to buy the bot. Tidal market is where you can rent them.

**Comment by TheMole442** [comment]
> i was looking a stellar aio, i put the wrong page. That one is to buy the bot. Tidal market is where you can rent them.

Probably wouldn’t be a bad idea really, saves buying it outright and trying to backward work out a queue by pass

**Comment by orsxn** [comment]
im pretty sure all you have to do is paste the queue link into the module. So should work on any queue it queue.

**Comment by bizarrist** [comment]
Can I be added to the private repo please, so I can ask some questions that I don't want appearing on this public one?

**Comment by njs90** [comment]
I've been reading about into Stellar AIO and have seen it mentioned that the queue-it module can only be used around 'big drops'. There is a UK focussed service that is similar, and the website shows it being used for ticketmaster, but I can't see it on any of the rental sites.

**Comment by TheMole442** [comment]
I’ve seen a few from the UK but I don’t know if you would be able to rent them, does the accommodation sale on Thursday have the same queue system?

**Comment by njs90** [comment]
I think it will do, and there is the local sale on Sunday too.

**Comment by orsxn** [comment]
seems like the local sale went back to previous system

**Comment by bizarrist** [comment]
> > you can rent them from whop, i was maybe thinking renting one for resale as it a queue it module. only like £6 a day.
> 
> which one is it that you can rent ? I messaged one of the guys you linked earlier in the thread about pricing

Did the trail run cold on this?

**Comment by bizarrist** [comment]
Hi. Is anyone still monitoring this or working on anything?

**Comment by bizarrist** [comment]
> > you can rent them from whop, i was maybe thinking renting one for resale as it a queue it module. only like £6 a day.
> 
> which one is it that you can rent ? I messaged one of the guys you linked earlier in the thread about pricing

Did they ever reply?

**Comment by keat-s** [comment]
Haven't worked on anything since the original sale tbh mate. Think the private repo has gone cold as well.
